File: Pippel.Core/Page.fs
namespace Pippel.Core

type Page<'T> =
    { CurrentPage: int
      PageCount: int
      PageSize: int
      ItemsCount: int64
      Items: 'T seq }

module Page =

    let map func page =
        { Page.Items = page.Items |> Seq.map func
          CurrentPage = page.CurrentPage
          PageCount = page.PageCount
          PageSize = page.PageSize
          ItemsCount = page.ItemsCount }

    let fromSeq seq =
        let count = seq |> Seq.length

        { Page.Items = seq
          CurrentPage = 0
          PageCount = count
          PageSize = count
          ItemsCount = count |> int64 }

File: Pippel.Tyche.Application/Core/src/main/java/com/pippel/core/Page.kt
package com.pippel.core

data class Page<TModel : Any>(
    val currentPage: Int,
    val pageSize: Int,
    val itemsCount: Int,
    val items: List<TModel>
) {

    fun hasNext(): Boolean {
        return currentPage < itemsCount / pageSize
    }

    fun next(): Int? {
        return if (hasNext()) currentPage + 1 else null
    }

}
